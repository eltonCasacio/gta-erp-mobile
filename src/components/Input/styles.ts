import styled from 'styled-components/native';
import Theme from '../../styles/Theme';

export const Wrapper = styled.View`
  width: 270px;
  background-color: ${Theme.Colors.white};
  border-radius: ${Theme.BorderRadios.small};
`;

export const Text = styled.TextInput`
  color: ${Theme.Colors.primary};
  font-size: ${Theme.Font.great};
`;
