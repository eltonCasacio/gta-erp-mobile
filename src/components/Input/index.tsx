import React from 'react';
import * as S from './styles';

type InputProps = {
  placeholder?: string;
  onChangeText: any;
  passwordType?: Boolean;
};

const Input = ({
  placeholder = 'placeholder',
  onChangeText,
  passwordType = false,
}: InputProps) => {
  return (
    <S.Wrapper>
      <S.Text
        secureTextEntry={passwordType}
        placeholder={placeholder}
        onChangeText={(text) => onChangeText(text)}
      />
    </S.Wrapper>
  );
};

export default Input;
