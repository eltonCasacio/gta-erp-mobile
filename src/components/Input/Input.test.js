import React from 'react';
import renderer from 'react-test-renderer';

import Input from '.';

describe('<Input>', () => {
  test('Testing layout and style', () => {
    const tree = renderer.create(<Input />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
