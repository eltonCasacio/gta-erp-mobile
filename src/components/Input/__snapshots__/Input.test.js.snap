// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> Testing layout and style 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fafafa",
        "borderRadius": 3,
        "width": 270,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    placeholder="placeholder"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "#070E70",
          "fontSize": 22,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
